src/basic-syntax/slices.md

```diff
@@ -19,16 +19,18 @@ fn main() {
 
 * We create a slice by borrowing `a` and specifying the starting and ending indexes in brackets.
 
-* If the slice starts at index 0, Rust’s range syntax means we can drop the starting index. 
+* If the slice starts at index 0, Rust’s range syntax allows us to drop the starting index, meaning that `&a[0..a.len()]` and `&a[..a.len()]` are identical.
     
-* The same is true for the last index, so `&a[2..a.len()]` and `&a[2..]` are equal.
+* The same is true for the last index, so `&a[2..a.len()]` and `&a[2..]` are identical.
 
-* `s` is a reference to a slice of `i32`s. Notice that the type of `s` no longer mentions the array length. This allows us to performing computations on slices of different sizes.
+* To easily create a slice of the full array, we can therefore use `&a[..]`.
```

Comment:
I've always been wondering when `&a` becomes a slice and when you need to use `&a[..]` (or the more explicit `a.to_slice()`. Is that something we could explain here?

Reply:
I'm not super clear on the distinction myself, but I *think* &a becomes a slice if type elision would have made it a slice anyway (based on the target type), and it defaults to a reference if no other type information is found. I've never gone into the details and always just accepted what the compiler suggested, but it'd be great to have more clarity here from someone more experienced than me, I agree.

Reply:
Okay, then let's leave it as it is for now. Someone else can come and fill in the details :smile: